src/pages/TokenDetails/index.tsx

```diff
@@ -46,26 +60,81 @@ const Widget = styled.div`
 `
 export default function TokenDetails() {
   const { tokenAddress } = useParams<{ tokenAddress?: string }>()
-  const { data, error, loading } = useTokenDetailPageQuery(tokenAddress)
+  const [loadingDetails, setLoading] = useState(true)
+  const theme = useTheme()
+  const tokenSymbol = useToken(tokenAddress)?.symbol
+  setTimeout(() => {
+    setLoading(false)
+  }, 1000)
 
   let tokenDetail
   if (!tokenAddress) {
     // TODO: handle no address / invalid address cases
     tokenDetail = 'invalid token'
-  } else if (loading) {
+  } else if (loadingDetails) {
     tokenDetail = <LoadingTokenDetail />
   } else {
     tokenDetail = <TokenDetail address={tokenAddress} />
   }
 
+  /* network balance handling */
+  const { data } = useNetworkTokenBalances({ address: tokenAddress })
+  const { connector, chainId: connectedChainId } = useWeb3React()
+  const { label: connectedLabel } = getChainInfoOrDefault(connectedChainId)
+  const totalBalance = 4.3 // dummy data
+
+  const chainsToList = useMemo(() => {
+    let chainIds = [...L1_CHAIN_IDS, ...L2_CHAIN_IDS]
+    const userConnectedToATestNetwork = connectedChainId && TESTNET_CHAIN_IDS.includes(connectedChainId)
+    if (!userConnectedToATestNetwork) {
+      chainIds = chainIds.filter((id) => !(TESTNET_CHAIN_IDS as unknown as SupportedChainId[]).includes(id))
+    }
+    return chainIds
+  }, [connectedChainId])
+
+  const balancesByNetwork = data
+    ? chainsToList.map((chainId) => {
+        const amount = data[chainId]
+        const fiatValue = amount // for testing purposes
+        if (!fiatValue || !isChainAllowed(connector, chainId)) return null
+        const chainInfo = getChainInfo(chainId)
+        if (!chainInfo) return null
+        return (
+          <NetworkBalance
+            key={chainId}
+            logoUrl={chainInfo.logoUrl}
+            balance={'1'}
+            tokenSymbol={tokenSymbol ?? 'XXX'}
+            fiatValue={fiatValue.toSignificant(2)}
+            label={chainInfo.label}
+            networkColor={theme.textPrimary}
```

Comment:
networkcolor is a textprimary?

Reply:
HMMM this was correct before, i wonder if I merged and this got overwritten -- I'll go through everything again